Source/ApiForm.swift

```diff
@@ -100,18 +125,17 @@ public class Api<ModelType:Object where ModelType:ApiModel> {
     }
     
     public func updateFromResponse(response: ApiModelResponse<ModelType>) {
-        if let statusCode = response.rawResponse?.status {
-            self.status = ApiModelStatus(statusCode: statusCode)
-        }
-        
         if let responseObject = response.responseObject {
+            
             model.modifyStoredObject {
-                self.model.updateFromDictionary(responseObject)
+                if let shouldInsertOrUpdate = self.model.insertOrUpdate where shouldInsertOrUpdate{
```

Comment:
I like the idea, but the two branches here do the same thing really. `self.model.updateFromDictionary(responseObject)` will crash if the model itself is persisted in Realm anyway? I have a separate branch where I think I have solved the `duplicate entry for primary key` error that you mentioned in the documentation. https://github.com/erkie/apimodel/tree/model-array-transform-tests